### PR TITLE
Lint all the things!!!

### PR DESCRIPTION
**Pull request by justinfagnani**
Adds depcheck, eslint and tslint tasks to the gulpfile, and fixes all lint errors.

This PR is probably easies to review commit by commit, since I grouped each type of fix into a single commit.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,12 +64,15 @@
     "chai": "^3.5.0",
     "depcheck": "^0.6.3",
     "eslint": "^2.9.0",
```
Comment (azakus) [inline package.json]: redundant with gulp-eslint

Comment (justinfagnani) [inline package.json]: done

**Comment by azakus** [comment]
`npm run lint` should now just be `gulp lint` in `package.json`

**Comment by azakus** [comment]
LGTM!

**Comment by FredKSchott** [comment]
So fresh, so clean

awesome work Justin

**Pull request merged**